test/e2e/e2e.go

```diff
@@ -40,3 +44,79 @@ func SetupWithNamespace(t *testing.T, namespace string) *test.Clients {
 	}
 	return clients
 }
+
+// CreateNamespace creates a namespace to run the integration tests.
+func CreateNamespace(t *testing.T, clients *test.Clients, namespace string) {
+	_, err := clients.KubeClient.Kube.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+	if err != nil && apierrs.IsNotFound(err) {
+		// Create the namespace if not available
+		if _, err = clients.KubeClient.Kube.CoreV1().Namespaces().Create(&corev1.Namespace{ObjectMeta:
+			metav1.ObjectMeta{Name: namespace}}); err != nil {
+			t.Fatalf("Failed to create the namespace %s: %v", namespace, err)
+		}
+	}
+}
+
+// KnativeServingVerify verifies if the KnativeServing can reach the READY status.
+func KnativeServingVerify(t *testing.T, clients *test.Clients, names test.ResourceNames) {
```

Comment:
nit, but do any of these really need to be public?

Reply:
they can be private so far.